doc/developer/platform/ux.md

```diff
@@ -1,20 +1,902 @@
-## Materialize Platform: User Experience
+# Materialize Platform: User Experience
 
-⚠️ **WARNING!** ⚠️ This doc is a work in progress!
+## How to read this document
 
-The intended user experience drives some of the concepts that we'll want to reflect in our code.
-The intended user experience in turn reflects the practical realities of the tools we have access to.
+⚠️ **WARNING!** ⚠️ This document is a work in progress! While the underlying
+architecture is firm, the particular mapping from architecture to user-facing
+concepts described in this document is still up for debate.
 
-The coarsest level of granularity of Materialize Platform is the ACCOUNT.
-Within an ACCOUNT there may be multiple REGIONs, which correspond roughly to availability zones.
-A REGION is the unit within which it is reasonable to transfer data, and across which that expectation doesn't exist.
+This document is a proposal for how we expose the
+[architecture](architecture.md) of Materialize Platform to users. The intended
+user experience drives some of the concepts that we'll want to reflect in our
+code. The intended user experience in turn reflects the practical realities of
+the tools we have access to.
 
-Within each REGION there may be multiple CLUSTERs, which correspond roughly to timely dataflow instances.
-A CLUSTER contains indexes and is the unit within which it is possible to share indexes.
+Terms in this document that represent key user-facing concepts are introduced
```

Comment:
This is the most contentious topic of the document for me (with the exception of region/environment that is explained below). What is the rationale for inventing two terms for the same thing? My take is that the concepts are already complex enough and we'll have to become fluent using the user terms when we talk to users anyway. So if there is no qualitative difference (like the multiplicity different between singular region/multiple environments) we should strive for a single term used everywhere.

Reply:
As part of the reason, the goal was to allow technical discussions to proceed without locking down later-binding product decisions. E.g. the one that precipitated this was REGION v ENVIRONMENT v STORAGE INSTANCE. From the architectural side, STORAGE has a quantum of sharing, and whether it is 1:1 with cloud regions, or m:1, or whatever, .. isn't something that the architecture doc should be locking down (at least, I don't have an opinion other than that we should build STORAGE so that it can create independent INSTANCEs).

It might be that we can race forward with the product discussions and determine the user-facing terms fast enough that the user-oblivious terms don't have to stick, and I'm ok with that.